client.go

```diff
@@ -841,15 +851,13 @@ func (cl *Client) receiveHandshakes(c *PeerConn) (t *Torrent, err error) {
 		err = errors.New("connection does not have required header obfuscation")
 		return
 	}
-	ih, err := cl.connBtHandshake(c, nil)
-	if err != nil {
-		err = xerrors.Errorf("during bt handshake: %w", err)
+	if ih, err := cl.connBtHandshake(c, nil); err == nil {
```

Comment:
I think leave this block the way it was.

Reply:
I tend to try to reduce the scope of variables whenever possible -- I think I was trying to do that for `ih` here.

I guess I could invert the if the statement to achieve the same effect of returning the error first.